ateam_kenobi/src/plays/wall_play.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright 2023 A Team
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include <CGAL/point_generators_2.h>
+
+#include <ateam_common/robot_constants.hpp>
+#include "wall_play.hpp"
+#include "robot_assignment.hpp"
+#include "types/robot.hpp"
+#include "skills/goalie.hpp"
+#include "play_helpers/available_robots.hpp"
+namespace ateam_kenobi::plays
+{
+std::vector<ateam_geometry::Point> get_equally_spaced_points_on_segment(
+  ateam_geometry::Segment & segment, int num_points)
+{
+  std::vector<ateam_geometry::Point> points_on_segment;
+
+  if (num_points == 1) {
+    points_on_segment.push_back(segment.vertex(0));
+  }
+
+  auto source = segment.vertex(0);
+  auto target = segment.vertex(1);
+
+  ateam_geometry::Point spacing = ateam_geometry::Point(
+    // source.x() + target.x() / (num_points - 1),
+    0,
+    CGAL::approximate_sqrt((source - target).squared_length()) / (num_points - 1)
+  );
+
+  for (int i = 0; i < num_points; ++i) {
+    auto segment_point = ateam_geometry::Point(
+      // source.x() + (spacing.x() * i),
+      target.x(),
+      target.y() + (spacing.y() * i)
+      // both of these were source in the last nothing else changed
+    );
+
+    points_on_segment.push_back(segment_point);
+    // WHAT THE ACTUAL FUCK
+  }
+  return points_on_segment;
+}
```

Comment:
I think we could replace this with the logic that's used in the penalty and stop plays to form a pattern instead

Reply:
CGAL is hard :(